### ISSUE TITLE
RFC: No longer check in iOS app build number

### ISSUE THREAD

**Issue by theospears** [beeminder/BeeSwift#571]
## TL;DR
Git history contains lots of uninteresting "Version Bump" commits reflecting each time we do a Testflight build. Remove the need for these commits by no longer checking in the build number.

## Problem
The BeeSwift commit history is reasonably clean. One exception to this is Version Bump commits. These are generated by fastlane each time an iOS build is generated. They
* Add clutter (I never care about them when looking at history)
* Require direct pushes to `master` as part of normal workflows (risky!)
* Are not particularly convenient when trying to map particular commits to versions.

They do serve one benefit, which is to provide a record of the code version for each fastlane release.

### 1. Remove the build number from code. The checked in version will always be 0
This will remove the need for any commit to ever update it. It means that locally built versions will not have a useful build number, but we already can't gaurantee that build number is particularly meaningful, so this doesn't seem like a big loss.

### 2. Modify fastlane not to check in build changes
Today fastlane will
(a) check there are no untracked changes
(b) Work out and apply the correct build number
(c) Commit that change
(d) Add a tag
(d) build and submit

Instead we will change it to
(a) check there are no untracked changes
(b) Work out and apply the correct build number
(c) build and submit
(d) Add a tag (of HEAD, not including uncommitted changes)
(e) Reset all version number related changes

It will remain possible to map particular releases to code versions using the same tags that exist today.

**Comment by krugerk** [comment]
If I am understanding this correctly, it might also imply setting the build for local dev builds to the max int would be desirable. Then a local build could replace the already installed version, right?

**Comment by theospears** [comment]
> If I am understanding this correctly, it might also imply setting the build for local dev builds to the max int would be desirable. Then a local build could replace the already installed version, right?

I was leaning towards 0, although max int would also work. I think local builds are happy to replace testflight builds regardless of version numbers.
